gnovm/pkg/gnolang/alloc.go

```diff
@@ -133,6 +144,14 @@ func (alloc *Allocator) Allocate(size int64) {
 			}
 		}
 	}
+	if alloc.bytes > alloc.maxBytesUsed {
+		if alloc.gasMeter != nil {
+			change := alloc.bytes - alloc.maxBytesUsed
+			alloc.gasMeter.ConsumeGas(overflow.Mulp(change, GasCostPerByte), "memory allocation")
+		}
+
+		alloc.maxBytesUsed = alloc.bytes
+	}
```

Comment:
so it is a linear model, we can work in future to add some the quadratic cost part.

Reply:
Yes, currently it's linear for simplicity and predictability. Adding quadratic (or other non-linear) cost models in the future could help capture more realistic scenarios